lib/widgets/category_card.dart

```diff
@@ -14,86 +14,237 @@
 // limitations under the License.
 //
 
-import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
 import 'package:arcgis_maps_sdk_flutter_samples/models/category.dart';
 import 'package:flutter/material.dart';
 
-/// A card widget that displays a SampleCategory.
-class CategoryCard extends StatelessWidget {
+class CategoryCard extends StatefulWidget {
   const CategoryCard({
     required this.category,
     required this.onClick,
+    required this.index,
     super.key,
   });
 
   final SampleCategory category;
-  final VoidCallback onClick;
+  final void Function(Offset tapPosition) onClick;
+  final int index;
+
+  @override
+  State<CategoryCard> createState() => _CategoryCardState();
+}
+
+class _CategoryCardState extends State<CategoryCard>
+    with SingleTickerProviderStateMixin {
+  double _scale = 1;
+  bool _iconPressed = false;
```

Comment:
Normally we would say `var _scale = 1` and `var _iconPressed = false`, and let the type be inferred.